datradito/DanielAgustin-Tradito-web
# Code Challenge Solution

### Pull Request: Code Challenge Solution

#### PR Description

**What does this PR do?**

- [x] Adds a new feature (Code Challenge Solution)
- [ ] Adds or improves tests

**How should this be manually tested?**

1. Clone the repo and checkout the `development` branch.
2. Run `npm install` or `yarn install` to install dependencies for both frontend and backend.
3. Start the backend server by navigating to the `back` directory and running `npm run start`.
4. Start the frontend by navigating to the `front` directory and running `npm run dev`.
5. Open your browser and navigate to `http://localhost:3000`.
6. Perform any manual tests to verify the features of the code challenge.

**Context**

This PR contains the complete solution for the given code challenge. It includes a backend built with NestJS and a frontend built with Next.js. The application allows users to fetch and display posts. It also includes tests and a theme toggle feature.

**What are the relevant tickets?**

(Code Challenge Submission)

**Screenshots**

<img width="1280" alt="image" src="https://github.com/datradito/DanielAgustin-Tradito-web/assets/19674406/5c8a8e96-db05-4de5-a70f-419c3b0dfe4f">

<img width="1278" alt="image" src="https://github.com/datradito/DanielAgustin-Tradito-web/assets/19674406/12697fc2-04df-4bc1-95db-362b45addc74">




**Questions:**
